test/e2e/daemon_set.go

```diff
@@ -245,15 +276,17 @@ func testDaemonSets(f *Framework) {
 		return false, err
 	})
 	Expect(err).NotTo(HaveOccurred())
-	Expect(len(newNode.Labels)).To(Equal(1))
+	daemonSetLabels, _ := separateDaemonSetNodeLabels(newNode.Labels)
+	Expect(len(daemonSetLabels)).To(Equal(1))
 	err = wait.Poll(retryInterval, retryTimeout, checkDaemonPodOnNodes(f, complexLabel, []string{newNode.Name}))
 	Expect(err).NotTo(HaveOccurred(), "error waiting for daemon pods to be running on new nodes")
 
 	By("remove the node selector and wait for daemons to be unscheduled")
 	newNode, err = nodeClient.Get(newNode.Name)
 	Expect(err).NotTo(HaveOccurred(), "error getting node")
-	newNode.Labels = map[string]string{}
 	err = wait.Poll(updateRetryPeriod, updateRetryTimeout, func() (bool, error) {
+		newNode, err = nodeClient.Get(newNode.Name)
+		setDaemonSetNodeLabels(newNode, map[string]string{})
 		newNode, err = nodeClient.Update(newNode)
```

Comment:
Sorry, same holds through for Node. Just do an unconditionalUpdate and the apiserver will not do a resource version check.
https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/node/strategy.go#L86